src/components/TodoApp.tsx

```diff
@@ -0,0 +1,221 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import {
+  FC,
+  useEffect,
+  useContext,
+  useRef,
+  useState,
+} from 'react';
+import { Route, Routes } from 'react-router-dom';
+import {
+  getTodos,
+  addTodo,
+  deleteTodo,
+  updateTodo,
+} from '../api/todo';
+import { AuthContext } from './AuthContext';
+import { Footer } from './Footer';
+import { Header } from './Header';
+import { TodoList } from './TodoList';
+import { ErrorNotification } from './ErrorNotification';
+import { Status } from '../types/Filter';
+import { useLocalStorage } from '../utils/useLocalStorage';
+import { Todo } from '../types/Todo';
+import { User } from '../types/User';
+import { Error } from '../types/Error';
+
+export const TodoApp: FC = () => {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const user = useContext<User | null>(AuthContext);
+  const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [localTodos, setLocalTodos] = useLocalStorage('todos', []);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
```

Comment:
Interesting decision to show user todo immediately after adding. It's sth remind optimistic updates

> Optimistic updates (a.k.a. Optimistic UI) is a pattern that we can especially see in frontend/UX development where different actions update instantly on the visual interface without waiting for confirmation from the server. 

Usually it's uses `todos` array directly, as I know, but I liked your try)